vue-app/src/plugins/Web3/index.ts

```diff
@@ -0,0 +1,99 @@
+import { LOGIN_MESSAGE, User } from '@/api/user'
+import { sha256 } from '@/utils/crypto'
+import { Web3Provider } from '@ethersproject/providers'
+import MetamaskConnector from './connectors/MetamaskConnector'
+import WalletConnectConnector from './connectors/WalletConnectConnector'
+
+export type Wallet = 'metamask' | 'walletconnect'
+
+const connectors: Record<Wallet, any> = {
+  metamask: MetamaskConnector,
+  walletconnect: WalletConnectConnector,
+}
+
+export default {
+  install: async (Vue) => {
+    const plugin = new Vue({
```

Comment:
Putting everything in a Vue instance for reactivity purposes. https://vuejs.org/v2/guide/reactivity.html